### PR TITLE
Add support for blue/green deployments

### PR DESCRIPTION
**Pull request by flupke**
- webhook certificate name is tied to release names
- add flags to control the deployment of the Role and ClusterRole separately

### CODE REVIEW

**Review by liyinan926 — APPROVED** [review]
Can you do a rebase? Thanks!

**Comment by flupke** [comment]
> Can you do a rebase? Thanks!

Done. Please note that the chart comes with a breaking change and has been updated to 2.0.0

**Comment by liyinan926** [comment]
> > Can you do a rebase? Thanks!
> 
> Done. Please note that the chart comes with a breaking change and has been updated to 2.0.0

Is it possible to keep the variable `rbac. create` and use an boolean `or` to check both `create` and the new variables? In the meantime, we can mark `create` as being deprecated.

**Comment by flupke** [comment]
Ok that's better indeed! how would I mark `create` deprecated?

**Comment by liyinan926** [comment]
> Ok that's better indeed! how would I mark `create` deprecated?

Probably just update the table in https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/master/charts/spark-operator-chart/README.md.

**Comment by flupke** [comment]
- Chart in 1.0.10 and rbac.create back compatibility: b933c7047b88406516
- Document rbac.create deprecation in charts/README (used `make helm-docs` for this, which also updated a bunch of other entries): 57c6c351643771a085

File: charts/spark-operator-chart/README.md [hunk 1]
```diff
@@ -60,24 +60,27 @@ The command removes all the Kubernetes components associated with the chart and
...
+| podMonitor.enable | bool | `false` | If enabled, a pod monitor for operator's pod will be submitted. Note that prometheus metrics should be enabled as well. |
+| podMonitor.jobLabel | string | `"spark-operator-podmonitor"` | The label to use to retrieve the job name from |
 | podMonitor.labels | object | `{}` | Pod monitor labels |
-| podMonitor.jobLabel | string | `spark-operator-podmonitor` | The label to use to retrieve the job name from |
-| podMonitor.podMetricsEndpoint.scheme | string | `http` | Prometheus metrics endpoint scheme |
-| podMonitor.podMetricsEndpoint.interval | string | `5s` | Interval at which metrics should be scraped |
+| podMonitor.podMetricsEndpoint | object | `{"interval":"5s","scheme":"http"}` | Prometheus metrics endpoint properties. `metrics.portName` will be used as a port |
 | podSecurityContext | object | `{}` | Pod security context |
-| rbac.create | bool | `true` | Create and use `rbac` resources |
+| rbac.create | bool | `false` | DEPRECATED - use createRole and createClusterRole |
```
Comment (liyinan926) [inline charts/spark-operator-chart/README.md:81]: Nit: add ` around createRole and createClusterRole

Comment (flupke) [reply charts/spark-operator-chart/README.md]: Done in daeda76

**Review by liyinan926 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**